lib/IRGen/GenStruct.cpp

```diff
@@ -290,7 +291,46 @@ namespace {
       }
     }
   };
-  
+
+  class ClangRecordAddressOnlyTypeInfo final
+      : public StructTypeInfoBase<ClangRecordAddressOnlyTypeInfo, FixedTypeInfo,
```

Comment:
Can someone more familiar with this part of IRGen confirm that `FixedTypeInfo` is what should go here?

Reply:
Should we use `IndirectTypeInfo` as the base type?

Reply:
You mean use `IndirectTypeInfo` instead of `StructTypeInfoBase`? Even address-only types should be imported as structs, so I think we should keep `StructTypeInfoBase`. `IndirectTypeInfo` takes two template parameters so, I think one of those would still have to be `FixedTypeInfo` or something.